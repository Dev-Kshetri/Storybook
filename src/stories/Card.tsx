import { CardComponent, CardComponentProps } from "./Card.styles";
import { FunctionComponent, ReactNode } from "react";


export interface CardProps extends CardComponentProps {
    size: "small" | "medium" | "large" | "auto";
    children?: ReactNode;
}


export const Card: FunctionComponent<CardProps> = ({ size = "small", children, borderColor = "#fff" }) => {
    return (
        <CardComponent className={`size-${size}`} borderColor={borderColor}>{children}</CardComponent>
    )
}
