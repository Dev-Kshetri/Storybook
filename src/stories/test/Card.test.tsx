import { render } from "@testing-library/react";
import { Card } from "../Card";


describe('<Card> Test', () => {

    it('should match snapshot and have given classes', () => {


        //Card size auto
        const { container: containerAuto } = render(<Card size="auto" borderColor="" />);
        expect(containerAuto).toMatchSnapshot();
        expect(containerAuto.getElementsByClassName("size-auto").length).toBe(1);

        //Card size large
        const { container: containerLarge } = render(<Card size="large" borderColor="" />);
        expect(containerLarge).toMatchSnapshot();
        expect(containerLarge.getElementsByClassName("size-large").length).toBe(1);


        //Card size small
        const { container: containerMedium } = render(<Card size="medium" borderColor="" />);
        expect(containerMedium).toMatchSnapshot();
        expect(containerMedium.getElementsByClassName("size-medium").length).toBe(1);



        //Card size auto
        const { container: containerSmall } = render(<Card size="small" borderColor="" />);
        expect(containerSmall).toMatchSnapshot();
        expect(containerSmall.getElementsByClassName("size-small").length).toBe(1);

    })

})