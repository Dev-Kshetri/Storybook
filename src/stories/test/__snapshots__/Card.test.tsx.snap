// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> Test should match snapshot and have classes 1`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-auto css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have classes 2`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-large css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have classes 3`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-medium css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have classes 4`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-small css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have gi 1`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-auto css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have gi 2`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-large css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have gi 3`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-medium css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have gi 4`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-small css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have given classes 1`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-auto css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have given classes 2`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-large css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have given classes 3`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-medium css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;

exports[`<Card> Test should match snapshot and have given classes 4`] = `
<div>
  <div
    bordercolor=""
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bczRLJ lftjCK size-small css-bhp9pd-MuiPaper-root-MuiCard-root"
  />
</div>
`;
